man/fifelse.Rd

```diff
@@ -0,0 +1,35 @@
+\name{fifelse}
+\alias{fifelse}
+\title{ Fast ifelse }
+\description{
+\code{data.table::fifelse} is comparable to \code{base::ifelse}, \code{dplyr::if_else} and  \code{hutils::if_else}.
+Like these functions, it returns a value with the same length as \code{test} and filled with value from \code{yes} or \code{no}.
+This function is mostly written in C for speed and unlike \code{base::ifelse} the output type is consistent with those of \code{yes} and \code{no}.
+}
+\usage{
+  fifelse(test, yes, no)
+}
+\arguments{
+  \item{test}{ A logical vector }
+  \item{yes, no}{  Values to return depending on \code{TRUE}/\code{FALSE} element of \code{test}. They must be the same type and be either length \code{1} or the same length of \code{test}. Attributes are copied from \code{yes} to the output.}
+}
+\value{
+A vector of the same length as \code{test} and attributes as \code{yes}. Data values are taken from the values of \code{yes} and \code{no}.
+}
+\examples{
+### Example 1
+dates <- as.Date(c("2011-01-01","2011-01-02","2011-01-03","2011-01-04","2011-01-05"))
+ifelse(dates == "2011-01-01", dates - 1, dates)
+# [1] 14974 14976 14977 14978 14979
+data.table::fifelse(dates == "2011-01-01", dates - 1, dates)
```

Comment:
i wouldn't use `data.table::` in the examples for the package

Reply:
That is fine by me. I'll change it. Thanks.

Reply:
I did not finish the openmp version (will commit when I can) but as discussed it is faster for large vectors but slower for small one ( -1e2:1e2 < 0 ). It makes fifelse slower than all the other functions (i.e. dplyr::fi_else, hutils::if_else and base::ifelse ). fifelse works for huge vectors like -1e9L:1e9L < 0 and the others run out of memory.

Reply:
@MichaelChirico , please see branch **fifelse-openmp** in my repo. You will find a function called `fifelse2` which is using multi-threading.  I couldn't push the test/benchmark higher as my laptop is really old.

```
# fifelse2 uses 2 threads.

x   <- -1e2L:1e2L < 0
microbenchmark::microbenchmark(
   data.table::fifelse2(x, 1L, 0L),
   data.table::fifelse(x, 1L, 0L),
   times = 100L
 )
# Unit: microseconds
#                             expr    min      lq     mean median     uq     max neval
#  data.table::fifelse2(x, 1L, 0L) 67.138 71.2010 77.01662 72.698 75.691 418.221   100
#   data.table::fifelse(x, 1L, 0L) 14.112 14.7535 17.41797 18.816 19.671  26.514   100


x   <- -1e8L:1e8L < 0 # 763Mb
microbenchmark::microbenchmark(
   data.table::fifelse2(x, 1L, 0L),
   data.table::fifelse(x, 1L, 0L),
   times = 100L
 )
# Unit: milliseconds
#                             expr      min       lq     mean   median       uq      max neval
#  data.table::fifelse2(x, 1L, 0L) 472.9533 489.7613 497.1877 495.4631 500.5052 607.3494   100
#   data.table::fifelse(x, 1L, 0L) 639.5606 658.2875 663.7909 661.8020 666.5692 739.6911   100

```

Please let me know what you think. Thank you.